server/historian/packages/historian-base/src/routes/index.ts

```diff
@@ -52,34 +53,39 @@ export function create(
 				restTenantThrottlers,
 				cache,
 				asyncLocalStorage,
+				tokenRevocationManager,
```

Comment:
The following comment is just a suggestion. If it makes the code more complicated to implement the way it is suggested then please ignore it.

node.js request pipeline is composed of a lot of middleware as described at https://www.turing.com/kb/building-middleware-for-node-js.
Is it possible to create a auth middleware at the beginning of the pipeline which is executed for all routes? Then we don't have to modify any of these routes by adding the extra parameter tokenRevocationManager. In the auth middleware, if the token is not revoked we can add a new property in the req object something called tokenNotRevoked. In all the routes, there must be a common place where we check the user is authenticated/authorized. We can check over there if the tokenNotRevoked property is present in req object.

Reply:
I think it is a very good suggestion. There is one API that I didn't add this middleware: ping, because it doesn't have all the information like tenant id and document id. But I agree we should do an analysis to see if we can group auth together instead of in each API routes. This also applies to Alfred endpoint since we call verifyStorageToken in every API routes. I will create a work item to track it. Thanks